### PR TITLE
feat(cookbook): PSBTs multiple inputs

### LINKED ISSUE #26: feat: Cookbook on how to sign multiple inputs.
I am currently designing a stateless WASM wallet and using pure `rust-bitcoin` and `rust-bip39` only.
The cookbook was awesome to sign a transaction with a single input.
However, we also want to sign transactions with multiple inputs.
Maybe choosing which indexes of the `Vec<Txin>` to sighash/sign.

I might take this one if no one is up to (after I finish the implementation)

### PR DESCRIPTION
**Pull request by storopoli**
Adds examples on how to create, update, sign, and finalize PSBTs version 0 that deals with multiple inputs.

Examples are SegWit V0 and Taproot.

- [x] updates CI to capture all nested `*.md` files
- [x] PSBT introduction
- [x] SegWit V0
- [x] Taproot

Closes #26.

### CODE REVIEW

**Comment by storopoli** [comment]
Ready to review guys!

File: cookbook/tests/generate.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ cat << EOF > src/lib.rs
 extern crate doc_comment;
 EOF
 
-for doc in ../src/*.md
+for doc in ../src/**/*.md
```
Comment (storopoli) [inline cookbook/tests/generate.sh:10]: The new examples are in a nested dir inside `src/`.
That's why this change

File: cookbook/src/psbt.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Working with PSBTs
+
+PSBTs (Partially Signed Bitcoin Transactions) are a format for encoding partially signed transactions.
```
Comment (tcharding) [inline cookbook/src/psbt.md]:
```suggestion
The Partially Signed Bitcoin Transaction (PSBT) format specifies an encoding for partially signed transactions.
```

File: cookbook/src/psbt.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+They are used in the context of multisignature wallets, hardware wallets,
```
Comment (tcharding) [inline cookbook/src/psbt.md]:
```suggestion
PSBTs are used in the context of multisignature wallets, hardware wallets,
```

File: cookbook/src/psbt.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+and other use cases where multiple parties need to collaborate to sign a transaction.
+
+PSBTs (version 0) are defined in [BIP 174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki).
```
Comment (tcharding) [inline cookbook/src/psbt.md]:
```suggestion
PSBT version 0 is defined in [BIP 174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki).
```

**Review by tcharding — COMMENTED** [review]
I started reviewing but then realised, while struggling with reviewing code in either the markdown file or the rendered HTML, that for the other cookbook stuff I did a PR into `rust-bitcoin/bitcoin/examples` first.

Any chance you would be willing to do that also, then we can review the code like normal and copying it to the cookbook is just a matter of cut'n'pasting the relevant bits.

**Comment by storopoli** [comment]
> Any chance you would be willing to do that also, then we can review the code like normal and copying it to the cookbook is just a matter of cut'n'pasting the relevant bits.

Yes of course!
That was my original idea.
I just thought that since this is a VERY stripped down of
[`bitcoin/examples/ecdsa-psbt.rs`](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/bitcoin/examples/ecdsa-psbt.rs)
and [`bitcoin/examples/taproot-psbt.rs`](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/bitcoin/examples/taproot-psbt.rs),
that they would not be "worth" of an example in `rust-bitcoin`.

That's why I PR'd here only and first.
But yeah I'll work on a code + comments version of these in `rust-bitcoin`.
That's a great suggestion and thanks for encouraging it.

**Review by tcharding — APPROVED** [review]
ACK 81c90e62eef00d17e86530b7c1c940f85740fd6c
